test/src/box/DedicatedWorkerRegistry.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe('dedicated worker registry', function() {
+  var Registry;
+  var OutputBox;
+  var Context;
+
+  beforeEach(function() {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 2147483647;
+
+    foam.CLASS({
+      package: 'foam.box.DedicatedWorkerRegistry.Test',
+      name: 'MockRegistry',
+      extends: 'foam.box.BoxRegistry',
+
+      properties: [
+        {
+          class: 'Array',
+          of: 'FObject',
+          name: 'actions'
+        }
+      ],
+
+      methods: [
+        function register(name, service, localBox) {
+          this.actions.push({
+            action: 'register',
+            name: name,
+            service: service,
+            box: localBox
+          });
+
+          // Copied code...
+          name = name || foam.next$UID();
+
+          var exportBox = this.SubBox.create({ name: name, delegate: this.me });
+          exportBox = service ? service.clientBox(exportBox) : exportBox;
+
+          this.registry[name] = {
+            exportBox: exportBox,
+            localBox: service ? service.serverBox(localBox) : localBox
+          };
+
+          return this.registry[name].exportBox;
+        },
+        function unregister(name) {
+          this.actions.push({
+            action: 'unregister',
+            name: name
+          });
+        },
+        function send(msg) {
+          return new Promise(function(resolve, reject) {
+            resolve("This was sent from mock registry!");
+          });
+        }
+      ]
+    });
+
+    MockRegistry = foam.lookup('foam.box.DedicatedWorkerRegistry.Test.MockRegistry');
+    Registry = foam.lookup('foam.box.DedicatedWorkerRegistry');
+    Context = foam.lookup('foam.box.Context');
+    LogBox = foam.lookup('foam.box.LogBox');
+  });
+
+  it('should forward all registrations to delegate by default', function() {
+    var ctx = Context.create();
+    var mockRegistry = MockRegistry.create(null, ctx);
```

Comment:
Assume that `mockRegistry` is acting as the "default" registry throughout these tests